modin/experimental/cloud/rpyc_proxy.py

```diff
@@ -353,9 +489,13 @@ class DeliveringMixin:
     for method in methods:
 
         def wrapper(self, *args, __remote_conn__=conn, __method_name__=method, **kw):
-            args = tuple(__remote_conn__.deliver(x) for x in args)
-            kw = {k: __remote_conn__.deliver(v) for k, v in kw.items()}
-            return getattr(self.__remote_end__, __method_name__)(*args, **kw)
+            args, kw = __remote_conn__.deliver(args, kw)
+            cache = object.__getattribute__(self, "__remote_methods__")
+            try:
+                remote = cache[__method_name__]
+            except KeyError:
+                cache[__method_name__] = remote = getattr(remote_cls, __method_name__)
```

Comment:
You probably should do the same as in `__prepare__`

Reply:
fixed, thx